forge/csrc/graph_lib/node_types.hpp

```diff
@@ -304,6 +304,16 @@ class ConstantInputNode : public InputNode
     bool equivalent(const ConstantInputNode *other) const;
 };
 
+enum class OutputType
+{
+    // Internal is used for outputs that are not exposed directly to the user, but used and handled internally for i/o
+    // between different graphs, e.g. passing intermediate values from forward graph to the backward graph.
+    Internal,
```

Comment:
Do we have any other use case (we know of) next to the passing gradients for bwd pass? 

Having Internal/External output types is a bit fuzzily. For example, having specific output types like "Gradients" can make it cleaner. However, I'm not certain if there are other special use cases that better fit into internal/external usage.

Reply:
Currently we are using it for passing intermediates from `fwd` to `bwd` programs. In future, for example, it will be also used for passing gradients from the `bwd` to the optimizer program. I'm not sure if differentiating between specific "types of things" we are passing between different programs adds any value.

Maybe I should've chosen a better name for this, instead of `OutputType`? Something like, `OutputScope`, `OutputVisibility`, or something else?